DOC: many small fixes

elliotwaite opened the pull request:
Some suggestions for the docs.

If any of these suggestions are incorrect or unwanted, feel free to edit this PR as needed.

elliotwaite committed:
DOC: many small fixes

jakevdp commented on README.md:
```diff
@@ -63,7 +63,7 @@ from jax import grad, jit, vmap
   for W, b in params:
     outputs = jnp.dot(inputs, W) + b
-    inputs = jnp.tanh(outputs)
+    outputs = jnp.tanh(outputs)
```
This code is correct as written (the outputs of one iteration become the inputs of the next iteration)

elliotwaite replied:
Ah! I think my brain wasn't registering the for loop for some reason. Thanks for clarifying.

elliotwaite replied:
This has been reverted in the latest commit.

jakevdp commented on cloud_tpu_colabs/JAX_demo.ipynb:
```diff
@@ -252,7 +252,7 @@
     "  for W, b in params:\n",
     "    outputs = jnp.dot(inputs, W) + b\n",
-    "    inputs = jnp.tanh(outputs)\n",
+    "    outputs = jnp.tanh(outputs)\n",
```
Same here

elliotwaite replied:
This has been reverted in the latest commit.

jakevdp reviewed (changes requested):
Thanks! Looks great, aside from two code examples that were correct as written.

elliotwaite commented:
Thanks for the review. I've reverted those incorrect code changes.

jakevdp commented:
Thanks - could you squash the changes into a single commit?

jakevdp commented:
Also, we'll need to fix the CI failure (the ipynb and md versions of `Writing_custom_interpreters_in_Jax` do not match).

See https://jax.readthedocs.io/en/latest/developer.html#update-notebooks

elliotwaite commented:
I fixed the ipynb/md issue.

Also, I think I squashed all the changes into the single commit titled "DOC: many small fixes", but this is my first time trying squashing so let me know if I did it incorrectly.

jakevdp commented:
Thanks! I still see 7 commits on the branch. You can find some info about squashing e.g. here: https://stackoverflow.com/q/5189560

Once you squash the commits locally, you can force push your branch and it will be updated in the pull request. Thanks!

jakevdp commented:
Since this touches so many files, it may also be a good idea to rebase on the main branch to pick up any conflicts that may have arisen since the initial commit.

jakevdp commented:
Sorry about all of this – I know it's not a normal requirement for github PRs in other packages, but it's necessary due to the system we have in place to keep internal & external repositories in sync.

jakevdp commented:
Thanks!

elliotwaite commented:
I think I finally figured out the squash and rebase. The force push tip was the key, thanks. And thanks for being patient with my GitHub skills. This has been a good opportunity for me to learn more about how Git works.

jakevdp commented:
Awesome! Thanks so much for the contribution 😁
